Add --list-devices option for BitBar

Cawllec opened the pull request:
## Goal

Allows a list of all device groups available to the given BitBar access key to be output to the console.

## Tests

Running manually, sample outputs include:

- No access key specified:
```
bundle exec maze-runner --farm=bb --list-devices
...
Listing BitBar device groups available requires a valid access key
```

- Invalid access key (i.e. No device groups found):
```
bundle exec maze-runner --farm=bb --access-key=foobar --list-devices
...
There are no device groups available for the given user access key
```

- Valid access key:
```
bundle exec maze-runner --farm=bb --access-key=<REDACTED> --list-devices
...
BitBar device groups available:
------------------------------
Group name   : ANDROID_10
Group ID     : <ID>
OS           : ANDROID
Device count : 8
------------------------------
Group name   : ANDROID_12
Group ID     : <ID>
OS           : ANDROID
Device count : 16
------------------------------
Group name   : ANDROID_13
Group ID     : <ID>
OS           : ANDROID
Device count : 36
```

twometresteve commented:
Nice change. It's already useful as it is, there are just a couple of tweaks that would be good before merging (sorting by name and removing some lines of output for easier reading). The others can we factored into separate PRs/tickets.

Cawllec commented:
Ok, with some small tweaks:
- Added extra guard against data being present but an empty array
- Device groups are now ordered by name
- Dropped group IDs from being output

The new output looks like:
```
BitBar device groups available:
------------------------------
Group name   : ANDROID_10
OS           : ANDROID
Device count : 8
------------------------------
Group name   : ANDROID_11
OS           : ANDROID
Device count : 1
------------------------------
Group name   : ANDROID_12
OS           : ANDROID
Device count : 16
------------------------------
```
